docs/sections/cn_workflows.md

```diff
@@ -414,6 +414,68 @@ either through the `--resource-manager/-r` option or through the config file.
 If neither of them are provided, the steps are run in the host machine 
 by default. 
 
+### Kubernetes
+
+Popper enables leveraging the compute and storage capabilities of the cloud by allowing running workflows on 
+Kubernetes clusters. One need to have access to the config file and ensure that any PersistentVolume is available
+inside the cluster. Popper takes care of the workflow execution from there.
+
+When a workflow is executed, Popper first creates a persistent volume claim and spawns an init pod and uses it to copy the workflow context (compressed in the form of a `.tar.gz` file) into the persistent volume. Then, popper teardowns this init pod and execute the steps of a workflow in separate pods of their own. After the execution of each step, the respective pods are deleted but the persistent volume claim is not deleted, so that it can reused by subsequent workflow executions.
+
+For running workflows on Kubernetes, some configuration options need to be passed to the kubernetes resource manager through the popper configuration file.
+
+#### Running on a cluster without shared storage
+
+**NOTE:** If your workflow needs to build an image from a `Dockerfile`, make sure you are logged in to dockerhub.
+
+1. Write a persistent volume defination similar to the one shown below.
+
+```bash
+$ cat<< EOF > pv.yaml
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: pv-hostpath
+  labels:
+    type: host
+spec:
+  persistentVolumeReclaimPolicy: Recycle
+  storageClassName: manual
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:
+    path: "/tmp"
+EOF
+```
+
+2. Then create the persistent volume.
+```bash
+$ kubectl apply -f pv.yaml
+```
+
+3. Write a configuration file similar to this.
```

Comment:
ideally we wouldn't need to add anything, and everything would be available in the kubctl config file.  So running in k8s would only require getting a k8s kubctl config file from an admin, and just run

```
popper run -r kubernetes
```

I think we have all we need, assuming a workflow is not building images, right?

Reply:
Yes, registry user need not be passed if our workflow doesnt build images.

Reply:
Also, as mentioned in the previous comment, if we can make local pv a default case, rather than an exception (which it is currently), i believe we will not require a config file.

Reply:
sounds good! Once #907 is merged, that would also help to not have to specify the registry here as well. The k8s workflows I am thinking are:
 1. no images built. All is needed is the kubectl config file (that includes the pv config).
 2. images built. Same as above, but before running a workflow users do `docker login` first, then specify images being built via the `build` attribute (as proposed in #907), and then do `docker run`. This also keeps the config out of the picture

Reply:
> images built. Same as above, but before running a workflow users do docker login first, then specify images being built via the build attribute (as proposed in #907), and then do docker run. This also keeps the config out of the picture

Yeah, correct. this should not require any config. thanks !